llvm/test/CodeGen/SPIRV/optimizations/add-check-overflow.ll

```diff
@@ -1,56 +0,0 @@
-; This test aims to check ability to support "Arithmetic with Overflow" intrinsics
```

Comment:
Not sure what the problem is with this test, but it's already covered by another?

Reply:
This relies on `CodeGenPrepare::combineToUSubWithOverflow` firing, however when encoding the int sizes we can do loop strength reduction before, which is preferable, but also means one no longer gets the overflow intrinsics inserted, hence the test is spurious. I think you mentioned elsewhere that it's risque to rely on specific optimisations, and since the correct lowering of the overflow intrinsics is already covered, this seems spurious, hence the removal.

Reply:
Right but it's relying on a non-guaranteed maybe-optimisation firing, as far as I can tell.

Reply:
@AlexVlx I'm strongly against deleting this test case.

Reply:
That is not the nature of this kind of test

Reply:
Can you please elaborate on "why"? Again, this relies on an opportunistic optimisation, that is not crucial / can be superseded by others. If the optimisation/transform is very important, then the way is not to just run -O3 anyway, but rather isolate codegenprepare and it's required analyses (which is doable, if a wee big painful). But I'd like to understand exactly what the intent behind the test was/is.

Reply:
The main objection is that the code base should switch from one stable state to another, without losing current coverage, stability, etc. When any of us is adding a feature that alters translation behavior it looks fair to expect that the same contributor is responsible for updating all impacted existing components. Applying this principle to the PR, I'd rather expect that you update existing test case (if it's required by the change), but not remove it.

Reply:
> Right but it's relying on a non-guaranteed maybe-optimisation firing, as far as I can tell.

The point is to test the optimization does work. The codegen pipeline is a bunch of intertwined IR passes on top of core codegen, and they need to cooperate

Reply:
@michalpaszkowski @VyacheslavLevytskyy I've restored the test, and retained the apparent intention of triggering combining in CodeGenPrepare. I do think there's a bit of an impedance mismatch in the conversation, but that might be just me: on one hand, there's nothing special about the pattern that results, and we already test for the correct generation of the intrinsics themselves. On the other, LSR does a better job here so it's not quite a case of a problem going away, but rather a more profitable optimisation becoming viable. Perhaps we could extend the test to cover this by way of, essentially, looking for the overflow intrinsics if LSR is off, and checking their absence when it's on, albeit that might introduce some added brittleness.

Reply:
The codegen prepare behavior is still backend code to be tested. You can just run codegenprepare as a standalone pass too (usually would have separate llc and opt run lines in such a test)